### PR TITLE
GTT-846: [Front-End] Modify Settings Page to Allow Edits to Topic Area Labels

### PR DESCRIPTION
**Pull request by ajmokotoff**
## Description

Added the front end changes to allow for the label "Topic Area" to be modified across the site.

## Testing

I added tests for the items added to the front end, I updated the snapshots, and confirmed that all other tests passed. I verified myself, by using testing various scenarios on the site itself.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/containers/__tests__/TopicareaSettings.test.tsx [hunk 1]
```diff
@@ -9,7 +9,18 @@ test("renders the title", async () => {
...
+    getByRole("heading", { name: "Topic areas (Ministries)" })
+  ).toBeInTheDocument();
+});
+
+test("renders the title with set value of topic areas", async () => {
+  const { getByRole } = render(<TopicareaSettings />, {
+    wrapper: MemoryRouter,
+  });
+  expect(
+    getByRole("heading", { name: "Topic areas (Ministries)" })
```
Comment (miguelpdiaz8) [inline frontend/src/containers/__tests__/TopicareaSettings.test.tsx]: As we discussed in the standup we need to use here the topicAreaLabel we just set and not longer use the ()

File: frontend/src/models/index.tsx [hunk 2]
```diff
@@ -152,6 +156,10 @@ export type Settings = {
   };
   updatedAt?: Date;
   navbarTitle?: string;
+  topicAreaLabels?: {
+    singular: string;
+    plural: string;
+  };
```
Comment (ferdingler) [inline frontend/src/models/index.tsx]: I don't think we should make this property optional. Remove the `?`.  I noticed that you added a default value in the `SetttingsProvider` file so we can assume that this property will always be present.

File: frontend/src/services/UtilsService.ts [hunk 3]
```diff
@@ -38,9 +38,22 @@ function getChartTypeLabel(chartType: string): string {
...
+  currentTopicArea: string | undefined
+): string {
+  return `Topic areas ${
+    currentTopicArea &&
+    currentTopicArea.toLowerCase() === defaultTopicArea.toLowerCase()
+      ? ""
+      : ` (${currentTopicArea})`
+  }`;
+}
+
```
Comment (ferdingler) [inline frontend/src/services/UtilsService.ts]: Nice solution for the title!

File: frontend/src/containers/__tests__/TopicareaSettings.test.tsx (hunk 1 above)
Comment (ferdingler) [reply frontend/src/containers/__tests__/TopicareaSettings.test.tsx]: That's true. I guess you no longer need the `determineTopicAreaString` function.

File: frontend/src/containers/TopicareaListing.tsx [hunk 4]
```diff
@@ -153,7 +153,7 @@ function TopicareaListing() {
               <span>
                 <Button
                   onClick={createTopicArea}
-                >{`Create ${EnvConfig.topicAreaLabel.toLowerCase()}`}</Button>
+                >{`Create new ${settings.topicAreaLabels?.singular.toLowerCase()}`}</Button>
```
Comment (ferdingler) [inline frontend/src/containers/TopicareaListing.tsx]: If we make the `topicAreaLabels` property not optional, we don't need these conditional `?` accessors here. I think it looks cleaner without them and it is safe to assume that the property will always be there given that it has default values in both the `SettingsProvider` and the backend endpoint.

**Review by ferdingler — CHANGES_REQUESTED** [review]

File: frontend/src/containers/__tests__/TopicareaSettings.test.tsx (hunk 1 above)
Comment (ajmokotoff) [reply frontend/src/containers/__tests__/TopicareaSettings.test.tsx]: Done!

File: frontend/src/models/index.tsx (hunk 2 above)
Comment (ajmokotoff) [reply frontend/src/models/index.tsx]: Agreed. Complete

File: frontend/src/services/UtilsService.ts (hunk 3 above)
Comment (ajmokotoff) [reply frontend/src/services/UtilsService.ts]: Removed, as we decided above

**Review by ferdingler — APPROVED** [review]

**Pull request merged**